### DISCUSSION TITLE
How do I "unpack" my watch face binary? And how safe is testing my watchface?

### DISCUSSION THREAD

**Discussion by gab51299** [ooflet/Mi-Create#39] [Questions-EN] [answered]
It's mentioned in the documentation that one way to check for errors is to unpack the watchface. Is this done in another tool? 

At what point can I brick my smart band? Will be getting a Xiaomi Band 9 as a gift for another person + watch faces, and really can't risk bricking the gift. The current faces ive made are very simple with just analog or digital times with basic PNGs, can this still cause errors?

**Comment by ooflet** [comment, marked as answer]
Thankfully Mi Band 9s have really good recovery, so there is little to no chance of bricking the device if you're not modifying firmware. If there is an error in the watchface, after 5 failed boots it'll factory reset itself. Chances are basic watchfaces are extremely unlikely to brick your watch, just take care not to overload the watch with too many widgets.

If you want to unpack watchfaces, you can download the [unpacker binary](https://github.com/ooflet/Mi-Create/blob/main/src/compiler/unpack.exe) or find it in your Mi Create install, either drag and drop the binary into the unpacker or use the command line. I will be adding a menu into the program to do this, but its not high priority right now.

**Reply by gab51299** [reply]
Thanks for the quick reply! Would you know about the Redmi Watch 2 Lite's recovery? Want to try on this older watch too

**Reply by ooflet** [reply]
I'm unsure about Redmi Watch 2 Lite's recovery but I'm pretty sure that watch doesn't have recovery especially because it's a Lite watch. I'd suggest being careful with overly complex watchfaces.

**Reply by xytunghoanh** [reply]
Can I specified model for unpacker from CLI?

**Reply by ooflet** [reply]
Unfortunately there is no way to specify a model to unpack through the CLI, you'll need to modify the unpacker to do so.
